src/coreclr/src/gc/gc.cpp

```diff
@@ -34925,9 +34925,19 @@ HRESULT GCHeap::Initialize()
     g_num_processors = GCToOSInterface::GetTotalProcessorCount();
     assert(g_num_processors != 0);
 
+    bool should_get_physical_mem = true;
     bool is_restricted;
     gc_heap::total_physical_mem = (size_t)GCConfig::GetGCTotalPhysicalMemory();
```

Comment:
What should happen if somebody sets a really high number via configuration, on 32-bit platforms? Should it get limited to the virtual address space size like for the configuration from the job object?

Reply:
if someone deliberately sets it to really high, then they will probably just run out of memory pretty quickly and then they should realize their mistake and fix it. we can also do a sanity check and return an error.